lib/devise/models/recoverable.rb

```diff
@@ -33,10 +33,13 @@ def self.required_fields(klass)
       # Update password saving the record and clearing token. Returns true if
       # the passwords are valid and the record was saved, false otherwise.
       def reset_password(new_password, new_password_confirmation)
-        self.password = new_password
-        self.password_confirmation = new_password_confirmation
-
-        save
+        if new_password.present?
+          self.password = new_password
+          self.password_confirmation = new_password_confirmation
+          save
+        else
+          errors.add(:password, :blank)
```

Comment:
@victor-am it'd be good to return `false` here, to keep the return value of `reset_password` the same (people might be relying on the previous truthy/falsy return value as to whether it worked or not)